### ISSUE TITLE
Add flag to startup which disables CLI

### ISSUE THREAD

**Issue by silvanheller** [vitrivr/cottontaildb#89]
Currently, if you want to deploy Cottontail via Docker, you need to add your own config-file which is an anti-pattern for docker deployment. 

I propose a simple solution: add a `--no-cli` flag, then the container can simply be started with a `--no-cli` flag if implemented properly, see https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#entrypoint

**Comment by ppanopticon** [comment]
This is not as straightforward to implement as it sound because we currently don't use any command line parser for the program arguments, since, so far we assume that there can only be one.

However, referring to #90: Would disabling the CLI by default in the config resolve this issue?

**Comment by sauterl** [comment]
Since we provide a Docker file, I would opt for yet another (imho neat) little trick:

Let's introduce the boolean env varialbe `COTTONTAIL_DOCKERISED` which our [Dockerfile](https://github.com/vitrivr/cottontaildb/blob/master/Dockerfile) defines. If such an env variable is present, the CLI is disabled.

**Comment by lucaro** [comment]
Wouldn't it be sufficient to change Cottontail.kt, line 50 as follows?

```kotlin
val config: Config = loadConfig(findConfigPathOrdered(args)).let { 
            if ("--no-cli" in args) {
                it.copy(cli = false)
            } else {
                it
            }
        }
```

**Comment by ppanopticon** [comment]
> Wouldn't it be sufficient to change Cottontail.kt, line 50 as follows?
> 
> ```kotlin
> val config: Config = loadConfig(findConfigPathOrdered(args)).let { 
>             if ("--no-cli" in args) {
>                 it.copy(cli = false)
>             } else {
>                 it
>             }
>         }
> ```

It would work but I wouldn't consider this a proper solution because we still assume and require the first argument to be either a path or either 'prompt' or 'cli'. Therefore,`./cottontail --no-cli /path/to/config` would, for example, fail.

**Comment by lucaro** [comment]
I see. So the semantics would need to be that no other config is loaded when a `--no-cli` flag is specified? Any other constellation would be somewhat redundant since if you wanted to specify a config, you would not need the flag in the first place?

@sauterl unless your proposed `COTTONTAIL_DOCKERISED` env variable would also have other effects, I'd strongly advise not to give it a name that is unrelated to what it does. Maybe `COTTONTAIL_HEADLESS` would be a better alternative?

**Comment by ppanopticon** [comment]
But maybe coming back to my initial suggestion: I have actually already thought about refactoring the CLI out of the main application completely. For example, `psql` for PostgreSQL is also a dedicated binary that can be started separately and used to connect to different instances. Wouldn't that be a solution?

I guess the only argument against this is lack of convenience during development...

**Comment by lucaro** [comment]
I would actually be in favor of having a dedicated CLI/UI application. The CLI is very difficult to use when cottontail is actively used since the default log output interferes with it anyway. It would probably also make it easier to start cottontail as a service.

**Comment by sauterl** [comment]
> @sauterl unless your proposed `COTTONTAIL_DOCKERISED` env variable would also have other effects, I'd strongly advise not to give it a name that is unrelated to what it does. Maybe `COTTONTAIL_HEADLESS` would be a better alternative?

Indeed, your name suggestion is much more descriptive and thus, fitting.

**Comment by ppanopticon** [comment]
This issues has been resolved with version `0.14.0` and the introduction of C(arrot)LI as dedicated CLI application.
